### PR TITLE
Optimistic responses

### PR DESCRIPTION
**Pull request by wtrocki**
Example of optimistic responses.

### CODE REVIEW

File: src/app/services/item.service.ts [hunk 1]
```diff
@@ -24,23 +24,42 @@ export class ItemService {
...
+      update: this.updateCacheOnEdit
+    });
   }
 
   deleteItem(item) {
-    return this.apollo.mutate<Task>({ mutation: DELETE_TASK, variables: { id: item.id } });
+    return this.apollo.mutate<Task>({
+      mutation: DELETE_TASK,
+      variables: { id: item.id },
+      update: this.updateCacheOnDelete
```
Comment (danielpassos) [inline src/app/services/item.service.ts]: Miss `,`

File: src/app/pages/home/home.page.ts [hunk 2]
```diff
@@ -26,16 +26,13 @@ export class HomePage implements OnInit {
...
-      console.log('error from query', error);
-      this.errors = error;
-    });
+
+  ionViewWillEnter() {
+    this.loadData();
+   }
+
+
+  ngOnInit() {
```
Comment (danielpassos) [inline src/app/pages/home/home.page.ts]: Miss `async`

File: src/app/services/item.service.ts [hunk 3]
```diff
@@ -24,23 +24,42 @@ export class ItemService {
   }
 
   getItems() {
-    return this.apollo.query<AllTasks>({ query: GET_TASKS });
+    return this.apollo.query<AllTasks>({
+      query: GET_TASKS,
+      fetchPolicy: 'network-only',
```
Comment (danielpassos) [inline src/app/services/item.service.ts]: It is not going to work when offline and we will get a network error.

**Comment by danielpassos** [comment]
I'm going to review it deep in all platform. Please don't merge it before I finish this.

**Comment by wtrocki** [comment]
No problem man! Going to drop subscriptions into this PR then.

**Comment by danielpassos** [comment]
I have found a problem on delete feature. TL;DR the delete works but the screen don't get a proper update.

https://www.dropbox.com/s/i4g41k7bdh3v7v0/aerogear_sync_js_ionic_optimistic_responses_web_delete.mov?dl=0

**Comment by danielpassos** [comment]
| -      |  Web               | Android            | iOS             |
|:------:|:------------------:|:------------------:|:---------------:|
| Insert | :heavy_check_mark: | :heavy_check_mark: | :grey_question: |
| Update | :heavy_check_mark: | :heavy_check_mark: | :grey_question: |
| Delete | :heavy_check_mark: | :heavy_check_mark: | :grey_question: |

**Pull request merged**